sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,1595 @@
+# Release notes Jul'24
+
+Release notes for commit range
+[d2817d6d317db1](https://github.com/intel/llvm/commit/d2817d6d317db1143bb227168e85c409d5ab7c82)
+...
+[ebb3b4a21b3b0e](https://github.com/intel/llvm/commit/ebb3b4a21b3b0e977f44434781729df7de83e436)
+
+## TODO
+
+commit https://github.com/intel/llvm/commit/9876e19f4ff387b35b0c98c7d62e5f50e6de187d
+    [SYCL][XPTI] 'queue_id' metadata feature refactoring (#13070)
+    bugfix?
+
+commit https://github.com/intel/llvm/commit/29b4d855fa1a378e89182795e0d368304c40c3f6
+    [SYCL][CUDA] Enable support of msvc math functions for nvptx target. (#14007)
+
+commit https://github.com/intel/llvm/commit/7a9d3b1e9483b69baa0b8c6f1097016efd52854c
+    [SYCL][NVPTX] Do not decompose SYCL functor unless necessary (#14434)
+
+commit https://github.com/intel/llvm/commit/38e663ecd37de513d8e31afdfdf245cf8c9d17f0
+    [SYCL] Declare __devicelib_assert_read only when fallback assert is enabled (#13241)
+    Is there any particular user-visible bug associated with this?
+
+commit https://github.com/intel/llvm/commit/4b993a7b32f7743980bce646765a1b427b0996b6
+    Revert "[SYCL][Driver] Link with sycl libs at link step of clang-cl -fsycl (#12793)" (#13326)
+    revert commit https://github.com/intel/llvm/commit/seems to be a part of a previous release
+
+commit https://github.com/intel/llvm/commit/4b14d706d93891cdb5b0e6a8d4b0b027c1d54ab8
+    [SYCL][DeviceSanitizer] Use -asan-constructor-kind=none to disable ctor/dtor (#13259)
+    was the bug really user-visible?
+commit https://github.com/intel/llvm/commit/2442ef047a4e9e9c135beed18a92029e1aad6cad
+    [DeviceSanitizer] Disable handling no return calls (#14652)
+    // bugfix?
+
++UR commit below
+
+## New Features
+
+### SYCL Compiler
+
+- Added `-fsycl-range-rounding` command line option which allows to
+  control range rounding feature. In comparison with previously available
+  `-fsycl-disable-range-rounding` command line option and
+  `__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__` macro the new flag also allows
+  to _force_ range rounding which will complete disable generation of
+  non-rounded kernels, thus improving binary size. intel/llvm#12715
+- Added `-fsycl-exp-range-rounding` command line option that enables
+  experimental range rounding mode in which range rounding is performed across
+  all dimensions. intel/llvm#12690
+- Added support for the so-called [new offloading model](https://github.com/intel/llvm/blob/ebb3b4a21b3b0e977f44434781729df7de83e436/sycl/doc/design/OffloadDesign.md).
+  It can be enabled by `--offload-new-driver` command line option and provides
+  a better infrastructure for us. In the future we expect to leverage that
```

Comment:
```suggestion
  It can be enabled by the `--offload-new-driver` command line option and provides
  an improved infrastructure. In the future we expect to leverage that
```

Reply:
Applied in 4dfea21f74ea84cd81abd20b140f02bb85bda622